src/lineup/internal/GeneralVisWrapper.ts

```diff
@@ -123,25 +75,61 @@ export class GeneralVisWrapper extends EventHandler {
     }
 
     updateCustomVis() {
-        const data = this.getAllData();
-        const colDescriptions = this.provider.getColumns();
-        //need some way to convert these to _ids.
-        const cols: any[] = [];
+        const ranking = this.provider.getFirstRanking();
+
+        const data = this.provider.viewRawRows(ranking.getOrder());
+
+        const cols: VisColumn[] = [];
 
         const selectedMap: { [key: number]: boolean } = this.getSelectionMap();
 
-        for(const c of colDescriptions.filter((d) => d.type === 'number' || d.type === 'categorical')) {
-            cols.push({
-                info: {
-                    name: c.label.replace(/(<([^>]+)>)/gi, ''),
-                    description: c.summary ? c.summary.replace(/(<([^>]+)>)/gi, '') : '',
-                    id: c.label + (<any> c)._id
-                },
-                values: data.map((d, i) => {
-                    return {id: d._id, val: d[(<any> c).column] ? d[(<any> c).column] : c.type === 'number' ? null : '--'};
-                }),
-                type: c.type === 'number' ? EColumnTypes.NUMERICAL : EColumnTypes.CATEGORICAL
+        const getColumnInfo = (column: Column): ColumnInfo => {
+            return {
+                name: column.getMetaData().label.replace(/(<([^>]+)>)/gi, ''),
+                description: column.getMetaData().summary.replace(/(<([^>]+)>)/gi, ''),
```

Comment:
Could you please add a short human-readable comment what the regex does? that makes it easier to understand. Thanks.

Reply:
added